SingularityService/src/main/java/com/hubspot/singularity/data/TaskManager.java

```diff
@@ -572,7 +573,10 @@ public SingularityCreateResult saveTaskHistoryUpdate(SingularityTaskHistoryUpdat
 
   @Timed
   public SingularityCreateResult saveTaskHistoryUpdate(SingularityTaskHistoryUpdate taskHistoryUpdate, boolean overwriteExisting) {
-    singularityEventListener.taskHistoryUpdateEvent(taskHistoryUpdate);
+    Optional<SingularityTask> task = getTask(taskHistoryUpdate.getTaskId());
```

Comment:
This is the only line I was hesitant about in the PR, but I still think it's a new improvement. With the previous version, we would be doing a write to zk here. Now we are doing a (mostly likely cached) read and a write to SNS/zk. The SNS update falls back to a zk write in case sns is down, so nothing should block here.